### PR TITLE
fix(archive-format): wrap index files as objects, drop dead QueryParams

### LINKED ISSUE #219: fix(archive-format): index files are bare JSON maps and can never gain a field
## Problem

`metadata.json` is a JSON object and can grow additively. But
`index.json.zst` and `watch-index.json.zst` are serialized as **bare
top-level maps** (`internal/capture/record.go:91`, `:106`).

That means adding a single index-level field — a schema marker, a count, a
checksum — requires bumping `CurrentFormatVersion`, which is exactly what the
compatibility guarantee in `docs/archive-format.md:60-121` promises not to do
casually.

## Also in scope

`Record.QueryParams` (`internal/capture/record.go:39`) is declared and has
**zero write sites anywhere in the repo**. It is a dead field about to be
frozen into a published format contract.

## Fix

- Wrap both indexes as `{"entries": {...}}` so they can evolve additively
- Delete `Record.QueryParams`, or populate it
- Do this **before** tagging, while it is still a free change

## Consider alongside

`?as=Table` / `?as=TableSchema` are stuffed into `api_path` as
pseudo-query-strings (`docs/archive-format.md:144`) where a `Kind`
discriminator belongs. `IndexEntry.Counts` encodes three states (non-list /
empty list / unknown) in two. Both are cheaper to change now than later.

## Acceptance criteria

- [ ] Both index files are objects with an `entries` key
- [ ] `CurrentFormatVersion` bumped once, deliberately, with the reader
      handling both shapes for the 1.x window
- [ ] `docs/archive-format.md` schema tables updated
- [ ] Golden fixtures regenerated deliberately

### PR DESCRIPTION
**Pull request by phenixblue**
## Summary

Second PR of [Phase 2](https://github.com/phenixblue/k8shark/issues/266), depends on #279 (merged) since the schema types now live in `internal/archive/format`.

`metadata.json` is a JSON object and can grow additively, but `index.json.zst` and `watch-index.json.zst` were bare top-level maps — adding a single index-level field (a schema marker, a count, a checksum) would require bumping `CurrentFormatVersion`, contradicting the additive-doesn't-bump compatibility promise in `docs/archive-format.md`.

## What changed

- Bump `CurrentFormatVersion` to **2** (deliberate, breaking, structural change).
- `Index` and `WatchIndex` gain `MarshalJSON`/`UnmarshalJSON`: writers always emit `{"entries": {...}}`; readers accept both the new wrapped shape **and** a version-1 archive's bare top-level map (unambiguous — a real api_path always starts with `/`, never equals `"entries"`). The Go-level type stays a plain map, so every existing caller across the repo (`idx[apiPath]`, `range idx`, `len(idx)`, ...) is unaffected — only the JSON (de)serialization changes, in one place.
- Delete `Record.QueryParams`: declared with zero write sites anywhere in the repo, about to be frozen into a published format contract.

## Golden fixtures

Added `golden-v2.kshrk` and `golden-v2-passphrase.kshrk` (pinned SHA-256 hashes, fail-loud-not-skip, same rigor as the v1 ones) to freeze the new format going forward. `golden-v1(-passphrase)` and their `sampleArchive`/`sampleEncryptedArchive` builder helpers are **deliberately left untouched** — they exist specifically to test that this build still reads a true historical v1-shaped archive, not whatever the current writer happens to produce. Regenerating them via `-update` would silently defeat that.

`TestArchiveFormatContract` now uses a new `sampleArchiveV2` helper (typed `format.Index`, matching what `internal/capture/engine.go` actually does) so it genuinely exercises and pins the current on-disk wrapped shape, rather than the raw `map[string]any` literals the old helper used (which bypassed `format.Index`'s `MarshalJSON` entirely).

## Docs

Updated `docs/archive-format.md`: `format_version` 2, the wrapped index/watch-index schema (with an explicit note on the version-1 bare-map compatibility shape), a new `watch-index.json.zst` section (it previously had no concrete schema example at all), and the "changing the format" contributor note (now points at `internal/archive/format`, and calls out freezing a new `golden-vN` pair rather than replacing existing ones).

## Acceptance criteria (from #219)

- [x] Both index files are objects with an `entries` key
- [x] `CurrentFormatVersion` bumped once, deliberately, with the reader handling both shapes for the 1.x window
- [x] `docs/archive-format.md` schema tables updated
- [x] Golden fixtures regenerated deliberately

## Test plan

- [x] `go build ./...`, `go vet ./...`, `gofmt -l .` clean, `go mod tidy` no diff, `golangci-lint run` clean.
- [x] `go test ./...` and `go test -race ./...` pass on all packages.
- [x] New unit tests in `internal/archive/format` cover: wrapped-shape marshal, round-trip, version-1 bare-map acceptance, empty-map edge cases (both shapes), and malformed-entry errors.

Closes #219

### CODE REVIEW

File: internal/archive/format/format.go [hunk 1]
```diff
@@ -91,9 +96,53 @@ type IndexEntry struct {
...
 
+// indexWire is the version-2+ on-disk shape of Index.
+type indexWire struct {
+	Entries map[string]*IndexEntry `json:"entries"`
+}
+
+// MarshalJSON always writes the version-2+ wrapped shape.
+func (idx Index) MarshalJSON() ([]byte, error) {
+	return json.Marshal(indexWire{Entries: map[string]*IndexEntry(idx)})
+}
```
[the remark this answers is not in the document]
Comment (phenixblue) [reply internal/archive/format/format.go]: Good catch — fixed. `Index.MarshalJSON` now normalizes a nil map to an empty one before marshaling, so `entries` is always `{}` rather than `null` for an empty/nil index. Added `TestIndex_MarshalJSON_NilIndexWritesEmptyObject`, verified it fails against the pre-fix code.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: internal/archive/format/format.go [hunk 2]
```diff
@@ -106,5 +155,45 @@ type WatchIndexEntry struct {
...
+
+// watchIndexWire is the version-2+ on-disk shape of WatchIndex.
+type watchIndexWire struct {
+	Entries map[string]*WatchIndexEntry `json:"entries"`
+}
+
+// MarshalJSON always writes the version-2+ wrapped shape.
+func (wi WatchIndex) MarshalJSON() ([]byte, error) {
+	return json.Marshal(watchIndexWire{Entries: map[string]*WatchIndexEntry(wi)})
+}
```
[the remark this answers is not in the document]
Comment (phenixblue) [reply internal/archive/format/format.go]: Fixed — same treatment as `Index.MarshalJSON`. Added `TestWatchIndex_MarshalJSON_NilWatchIndexWritesEmptyObject`.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: internal/archive/format/format_test.go [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+	}
+	for path, wantEntry := range want {
+		gotEntry, ok := got[path]
+		if !ok {
+			t.Errorf("entry %q missing after round trip", path)
+			continue
+		}
+		if gotEntry.APIPath != wantEntry.APIPath || len(gotEntry.Seqs) != len(wantEntry.Seqs) {
+			t.Errorf("entry %q = %+v, want %+v", path, gotEntry, wantEntry)
+		}
```
[the remark this answers is not in the document]
Comment (phenixblue) [reply internal/archive/format/format_test.go]: Fixed — the test now compares `Seqs`/`Counts` via `reflect.DeepEqual` (with distinct, non-symmetric values so a reordering couldn't hide behind coincidentally-equal-looking slices) instead of just comparing lengths.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
